onpremise/onpremise.go

```diff
@@ -0,0 +1,152 @@
+package onpremise
+
+import (
+	"fmt"
+	"github.com/51Degrees/device-detection-go/v4/dd"
+	"os"
+)
+
+type Engine struct {
```

Comment:
can Engine also be like in other languages - able to process evidence?  it should be a full wrapper over device detection, not just for data file updates...